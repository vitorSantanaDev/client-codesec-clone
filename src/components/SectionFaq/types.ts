import { ISectionFaqProps } from 'interfaces/sections'

export type IViewProps = ISectionFaqProps
