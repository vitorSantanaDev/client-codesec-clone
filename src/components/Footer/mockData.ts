export const mockDataFooterProps = {
  logo: {
    altText: 'logo codesec',
    urlImage: '/images/logo-2-codesec.svg'
  },
  footerContent: 'Todos od direitos reservados',
  links: [
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/instagram.svg'
      }
    },
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/facebook.svg'
      }
    },
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/youtube.svg'
      }
    }
  ]
}
