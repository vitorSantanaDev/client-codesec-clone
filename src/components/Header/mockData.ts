import { IViewProps } from './type'

export const mockHeaderProps: IViewProps = {
  logo: {
    urlImage: '/images/codesec-logo.svg',
    altText: 'Logo codesec'
  },
  button: {
    url: '/',
    buttonText: 'Inscreva-se'
  }
}
