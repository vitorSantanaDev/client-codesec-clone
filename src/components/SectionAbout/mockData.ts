export const mockSectionAboutProps = {
  subtitle: 'infos',
  title: 'About web series',
  description:
    'Aprenda PHP do Jeito Certo Utilizando 100% de Tecnologia Moderna e os Melhores Padr√µes de Projeto Nessa Webserie 100% Online e GRATUITA!',
  topics: [
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    }
  ],
  illustration: {
    urlImage: '/images/about-ilustration.svg',
    altText: 'Illustration about'
  }
}
