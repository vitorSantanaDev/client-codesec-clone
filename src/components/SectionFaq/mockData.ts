import { IViewProps } from './types'

export const mockDataSectionFaq: IViewProps = {
  subtitle: 'FAQ',
  title: 'Perguntas frequentes',
  questions: [
    {
      _id: '01',
      question: 'Question expample, uestion example?',
      answer:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis nunc convallis.',
      open: false
    },
    {
      _id: '02',
      question: 'Question expample, uestion example?',
      answer:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis nunc convallis.',
      open: false
    },
    {
      _id: '03',
      question: 'Question expample, uestion example?',
      answer:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis nunc convallis.',
      open: false
    },
    {
      _id: '04',
      question: 'Question expample, uestion example?',
      answer:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis nunc convallis.',
      open: false
    },
    {
      _id: '05',
      question: 'Question expample, uestion example?',
      answer:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis nunc convallis.',
      open: false
    },
    {
      _id: '06',
      question: 'Question expample, uestion example?',
      answer:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis nunc convallis. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis nunc convallis. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis nunc convallis.',
      open: false
    }
  ]
}
