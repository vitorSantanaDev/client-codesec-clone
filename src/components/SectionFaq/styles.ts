import styled, { css, keyframes } from 'styled-components'
import { Wrapper as Heading } from 'components/Heading/styles'
import { Wrapper as Subtitle } from 'components/Subtitle/styles'

type IAccordionProps = {
  showAnswer: boolean
}

const showAnswerAnimation = keyframes`
  from {
    max-height: 60px;
  }

  to {
    max-height: 200px;
  }
`
const closeAnswerAnimation = keyframes`
  from {
    max-height: 200px;
  }

  to {
    max-height: 60px;
  }
`

export const Wrapper = styled.section`
  ${({ theme }) => css`
    width: 100%;
    display: flex;
    justify-content: space-between;

    padding: ${theme.spacings.huge} 0;

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      justify-content: flex-start;
    }
  `}
`

export const HeadingAndSubtitleWrapper = styled.div`
  ${({ theme }) => css`
    > ${Subtitle} {
      margin-bottom: calc(${theme.spacings.small} - 0.8rem);
    }

    > ${Heading} {
      max-width: 384px;

      @media ${theme.media.lteMedium} {
        margin-bottom: ${theme.spacings.large};
      }
    }
  `}
`

export const AccordionsWrapper = styled.div`
  ${() => css``}
`

export const Accordion = styled.div<IAccordionProps>`
  ${({ theme, showAnswer }) => css`
    width: 100%;
    max-width: 696px;
    overflow: hidden;
    max-height: ${!showAnswer ? '60px' : '500px'};

    margin-bottom: ${theme.spacings.small};
    padding: ${theme.spacings.small} ${theme.spacings.medium};

    transition: transform 700ms ease-in-out;

    ${showAnswer
      ? css`
          animation: ${showAnswerAnimation} 700ms ease-in-out forwards;
        `
      : css`
          animation: ${closeAnswerAnimation} 700ms ease-in-out forwards;
        `}

    border-radius: 6px;
    background-color: ${theme.colors.black};

    &:hover {
      transform: translate3d(5px, 0px, 0px);
    }
  `}
`

export const AccordionHeader = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: baseline;
    justify-content: space-between;

    > ${Heading} {
      margin-bottom: calc(${theme.spacings.small} + 0.4rem);
      font-size: calc(${theme.font.sizes.small} + 0.2rem);

      @media ${theme.media.lteMedium} {
        font-size: calc(${theme.font.sizes.small} - 0.2rem);
      }
    }
  `}
`

export const AccordionButton = styled.button`
  ${({ theme }) => css`
    width: 24px;
    height: 24px;
    border: none;
    cursor: pointer;

    border-radius: 5px;
    background-color: ${theme.colors.blackLight};
  `}
`

export const AccordionDescription = styled.p<IAccordionProps>`
  ${({ theme, showAnswer }) => css`
    line-height: 150%;
    opacity: ${showAnswer ? 1 : 0};
    transition: all ease 600ms;
    color: ${theme.colors.gray7};
    font-size: ${theme.font.sizes.small};
  `}
`
