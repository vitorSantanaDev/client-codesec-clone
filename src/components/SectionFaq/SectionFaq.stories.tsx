import { ComponentStory, ComponentMeta } from '@storybook/react'
import SectionFaq from '.'
import { mockDataSectionFaq } from './mockData'

export default {
  title: 'SectionFaq',
  component: SectionFaq,
  args: {
    ...mockDataSectionFaq
  }
} as ComponentMeta<typeof SectionFaq>

export const Default: ComponentStory<typeof SectionFaq> = (args) => (
  <SectionFaq {...args} />
)
