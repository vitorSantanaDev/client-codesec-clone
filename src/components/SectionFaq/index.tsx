import { useState } from 'react'
import { ChevronUp, ChevronDown } from '@styled-icons/feather'

import Container from 'components/Container'
import Heading from 'components/Heading'
import Subtitle from 'components/Subtitle'
import { theme } from 'styles/theme/theme'
import * as S from './styles'

type ISectionFaqProps = {
  subtitle: string
  title: string
  questions: {
    _id: string
    question: string
    answer: string
    open: boolean
  }[]
}

const SectionFaq = ({ title, subtitle, questions }: ISectionFaqProps) => {
  const [questionsState, setQuestionsState] = useState(questions)

  const handleShowAnswer = (index: number) => {
    questionsState[index].open = !questionsState[index].open
    setQuestionsState([...questionsState])
  }

  return (
    <Container>
      <S.Wrapper>
        <S.HeadingAndSubtitleWrapper>
          <Subtitle>{subtitle}</Subtitle>
          <Heading>{title}</Heading>
        </S.HeadingAndSubtitleWrapper>
        <S.AccordionsWrapper>
          {questionsState.map((question, index) => (
            <S.Accordion showAnswer={question.open} key={question._id}>
              <S.AccordionHeader>
                <Heading size="small" as="h6">
                  {question.question}
                </Heading>
                <S.AccordionButton onClick={() => handleShowAnswer(index)}>
                  {question.open ? (
                    <ChevronUp
                      size={20}
                      color={
                        question.open
                          ? theme.colors.redLight
                          : theme.colors.white
                      }
                    />
                  ) : (
                    <ChevronDown
                      size={20}
                      color={
                        question.open
                          ? theme.colors.redLight
                          : theme.colors.white
                      }
                    />
                  )}
                </S.AccordionButton>
              </S.AccordionHeader>
              <S.AccordionDescription showAnswer={question.open}>
                {question.answer}
              </S.AccordionDescription>
            </S.Accordion>
          ))}
        </S.AccordionsWrapper>
      </S.Wrapper>
    </Container>
  )
}

export default SectionFaq
