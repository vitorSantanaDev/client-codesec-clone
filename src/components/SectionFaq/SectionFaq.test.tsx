import { screen } from '@testing-library/react'
import { renderTheme } from 'styles/theme/render-theme'
import SectionFaq from '.'
import { mockDataSectionFaq } from './mockData'

describe('<SectionFaq />', () => {
  it('should render subtitle and subtitle', () => {
    renderTheme(<SectionFaq {...mockDataSectionFaq} />)
    const heading = screen.getByRole('heading', {
      name: mockDataSectionFaq.title
    })
    const subtitle = screen.getByText(mockDataSectionFaq.subtitle)

    expect(heading).toBeInTheDocument()
    expect(subtitle).toBeInTheDocument()
  })
  it('should render the correct amount of questions', () => {
    renderTheme(<SectionFaq {...mockDataSectionFaq} />)
    const questions = screen.getAllByRole('heading', {
      name: mockDataSectionFaq.questions[0].question
    })
    const buttons = screen.getAllByRole('button')

    expect(questions).toHaveLength(6)
    expect(buttons).toHaveLength(6)
  })
})
