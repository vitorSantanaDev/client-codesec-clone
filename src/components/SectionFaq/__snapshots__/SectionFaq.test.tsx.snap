// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionFaq /> should match snapshot 1`] = `
.c11 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  width: 100%;
  margin: 0 auto;
  max-width: 1200px;
  padding: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  font-size: 5.6rem;
  color: white;
  font-weight: 600;
  font-family: 'Raleway',sans-serif;
}

.c9 {
  font-size: 1.6rem;
  color: white;
  font-weight: 600;
  font-family: 'Raleway',sans-serif;
}

.c4 {
  background-image: linear-gradient(243.46deg,#FA5770 16.66%,#E83752 83.34%);
  background-clip: text;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 700;
  -webkit-letter-spacing: 15%;
  -moz-letter-spacing: 15%;
  -ms-letter-spacing: 15%;
  letter-spacing: 15%;
  text-transform: uppercase;
  font-size: calc(1.6rem - 0.2rem);
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 5.6rem 0;
}

.c2 > .c3 {
  margin-bottom: calc(1.6rem - 0.8rem);
}

.c2 > .c5 {
  max-width: 384px;
}

.c7 {
  width: 100%;
  max-width: 696px;
  overflow: hidden;
  max-height: 60px;
  margin-bottom: 1.6rem;
  padding: 1.6rem 2.4rem;
  -webkit-transition: -webkit-transform 700ms ease-in-out;
  -webkit-transition: transform 700ms ease-in-out;
  transition: transform 700ms ease-in-out;
  -webkit-animation: hQFOSm 700ms ease-in-out forwards;
  animation: hQFOSm 700ms ease-in-out forwards;
  border-radius: 6px;
  background-color: #121111;
}

.c7:hover {
  -webkit-transform: translate3d(5px,0px,0px);
  -ms-transform: translate3d(5px,0px,0px);
  transform: translate3d(5px,0px,0px);
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c8 > .c5 {
  margin-bottom: calc(1.6rem + 0.4rem);
  font-size: calc(1.6rem + 0.2rem);
}

.c10 {
  width: 24px;
  height: 24px;
  border: none;
  cursor: pointer;
  border-radius: 5px;
  background-color: #1C1B1B;
}

.c12 {
  line-height: 150%;
  opacity: 0;
  -webkit-transition: all ease 600ms;
  transition: all ease 600ms;
  color: #A3A3A3;
  font-size: 1.6rem;
}

@media (max-width:768px) {
  .c6 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c9 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media (max-width:768px) {
  .c2 > .c5 {
    margin-bottom: 3.2rem;
  }
}

@media (max-width:768px) {
  .c8 > .c5 {
    font-size: calc(1.6rem - 0.2rem);
  }
}

<div
  class="c0"
>
  <section
    class="c1"
  >
    <div
      class="c2"
      data-aos="fade-down-right"
    >
      <span
        class="c3 c4"
        color="redGradient"
      >
        FAQ
      </span>
      <h1
        class="c5 c6"
        color="white"
      >
        Perguntas frequentes
      </h1>
    </div>
    <div
      class=""
    >
      <div
        class="c7"
        data-aos="fade-up-right"
      >
        <div
          class="c8"
        >
          <h6
            class="c5 c9"
            color="white"
          >
            Question expample, uestion example?
          </h6>
          <button
            class="c10"
          >
            <svg
              aria-hidden="true"
              class="c11"
              color="#FFFFFF"
              fill="none"
              focusable="false"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="6 9 12 15 18 9"
              />
            </svg>
          </button>
        </div>
        <p
          class="c12"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis nunc convallis.
        </p>
      </div>
      <div
        class="c7"
        data-aos="fade-up-right"
      >
        <div
          class="c8"
        >
          <h6
            class="c5 c9"
            color="white"
          >
            Question expample, uestion example?
          </h6>
          <button
            class="c10"
          >
            <svg
              aria-hidden="true"
              class="c11"
              color="#FFFFFF"
              fill="none"
              focusable="false"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="6 9 12 15 18 9"
              />
            </svg>
          </button>
        </div>
        <p
          class="c12"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis nunc convallis.
        </p>
      </div>
      <div
        class="c7"
        data-aos="fade-up-right"
      >
        <div
          class="c8"
        >
          <h6
            class="c5 c9"
            color="white"
          >
            Question expample, uestion example?
          </h6>
          <button
            class="c10"
          >
            <svg
              aria-hidden="true"
              class="c11"
              color="#FFFFFF"
              fill="none"
              focusable="false"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="6 9 12 15 18 9"
              />
            </svg>
          </button>
        </div>
        <p
          class="c12"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis nunc convallis.
        </p>
      </div>
      <div
        class="c7"
        data-aos="fade-up-right"
      >
        <div
          class="c8"
        >
          <h6
            class="c5 c9"
            color="white"
          >
            Question expample, uestion example?
          </h6>
          <button
            class="c10"
          >
            <svg
              aria-hidden="true"
              class="c11"
              color="#FFFFFF"
              fill="none"
              focusable="false"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="6 9 12 15 18 9"
              />
            </svg>
          </button>
        </div>
        <p
          class="c12"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis nunc convallis.
        </p>
      </div>
      <div
        class="c7"
        data-aos="fade-up-right"
      >
        <div
          class="c8"
        >
          <h6
            class="c5 c9"
            color="white"
          >
            Question expample, uestion example?
          </h6>
          <button
            class="c10"
          >
            <svg
              aria-hidden="true"
              class="c11"
              color="#FFFFFF"
              fill="none"
              focusable="false"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="6 9 12 15 18 9"
              />
            </svg>
          </button>
        </div>
        <p
          class="c12"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis nunc convallis.
        </p>
      </div>
      <div
        class="c7"
        data-aos="fade-up-right"
      >
        <div
          class="c8"
        >
          <h6
            class="c5 c9"
            color="white"
          >
            Question expample, uestion example?
          </h6>
          <button
            class="c10"
          >
            <svg
              aria-hidden="true"
              class="c11"
              color="#FFFFFF"
              fill="none"
              focusable="false"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="6 9 12 15 18 9"
              />
            </svg>
          </button>
        </div>
        <p
          class="c12"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis nunc convallis. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis nunc convallis. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis nunc convallis.
        </p>
      </div>
    </div>
  </section>
</div>
`;
